test(refactor): include VFS test cases in manual test plan file (#631)
opencloud-eu/desktop: test/manual/testplan.md

|----|----------------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|--------------------------------------------------------------------|-------------------------------|

> | 1  | Add account and connect to server            | 1. Create two files with some content inside a folder in the server <br>2. Add an account to the desktop client <br>3. Open local sync folder and check status of all resources (2 files and their parent folder)                                                                                                                                                                                                                      | All resources (2 files and their parent folder) are virtual, status: placeholder file (cloud icon)                                                                                                                                                                                                                                                                                                                                               | :construction: Win<br>:construction: macOS<br>:construction: Linux |                               |

Reviewer: VFS is for Windows only, right?
```suggestion
| 1  | Add account and connect to server            | 1. Create two files with some content inside a folder in the server <br>2. Add an account to the desktop client <br>3. Open local sync folder and check status of all resources (2 files and their parent folder)                                                                                                                                                                                                                      | All resources (2 files and their parent folder) are virtual, status: placeholder file (cloud icon)                                                                                                                                                                                                                                                                                                                                               | :construction: Win |                               |
```

> | 2  | Opening of file                              | 1. Create a file with some content in the server <br>2. Add an account to the desktop client <br>3. Open local sync folder and open that file <br>4. Create a new file locally in the local sync folder, edit, save and exit it                                                                                                                                                                                                        | Both files are physically available, status: full file (green unfilled circle with green check mark)                                                                                                                                                                                                                                                                                                                                             | :construction: Win<br>:construction: macOS<br>:construction: Linux |                               |

Reviewer: ```suggestion
| 2  | Open a file                              | 1. Create a file with some content in the server <br>2. Add an account to the desktop client <br>3. Open local sync folder and open that file <br>4. Create a new file locally in the local sync folder, edit, save and exit it                                                                                                                                                                                                        | Both files are physically available, status: full file (green unfilled circle with green check mark)                                                                                                                                                                                                                                                                                                                                             | :construction: Win<br>:construction: macOS<br>:construction: Linux |                               |
```

> | 2  | Opening of file                              | 1. Create a file with some content in the server <br>2. Add an account to the desktop client <br>3. Open local sync folder and open that file <br>4. Create a new file locally in the local sync folder, edit, save and exit it                                                                                                                                                                                                        | Both files are physically available, status: full file (green unfilled circle with green check mark)                                                                                                                                                                                                                                                                                                                                             | :construction: Win<br>:construction: macOS<br>:construction: Linux |                               |

Reviewer: use above terms
```suggestion
| 2  | Opening of file                              | 1. Create a file with some content in the server <br>2. Add an account to the desktop client <br>3. Open local sync folder and open that file <br>4. Create a new file locally in the local sync folder, edit, save and exit it                                                                                                                                                                                                        | Both files are physically available (Status: hydrated)                                                                                                                                                                                                                                                                                                                                            | :construction: Win<br>:construction: macOS<br>:construction: Linux |                               |
```

| 6  | Move file and directory locally              | 1. Create a file and a directory (with some files) inside the local sync root (some hydrated, some dehydrated) <br>2. Move (Cut + Paste) the file and the directory to a different folder inside the sync root <br>3. Also move a dehydrated folder from inside sync root to a location outside the sync root                                                                                                                          | Both hydrated and dehydrated items are moved accordingly on the server (i.e., the same structure appears on the server)                                                                                                                                                                                                                                                                                                                          | :construction: Win<br>:construction: macOS<br>:construction: Linux |                               |

> | 7  | Move file and directory outside sync root    | 1. Create a folder inside the sync root (which may be hydrated or dehydrated) <br>2. Move this folder to a location outside the sync root                                                                                                                                                                                                                                                                                              | The folder is removed on the server as well and no residual placeholder remains locally                                                                                                                                                                                                                                                                                                                                                          | :construction: Win<br>:construction: macOS<br>:construction: Linux |                               |

Reviewer: Case 6 has last step similar to 7 and can be removed from case 6

> | 8  | Move file/directory on the server            | 1. Within the server (via web UI) move a file and a directory (including some dehydrated files) from one location to another                                                                                                                                                                                                                                                                                                           | The local sync root reflects the move: both hydrated and dehydrated files show up in the new location locally; previous location cleared                                                                                                                                                                                                                                                                                                         | :construction: Win<br>:construction: macOS<br>:construction: Linux |                               |

Reviewer: ```suggestion
| 8  | Move file/directory on the server            | 1. Within the server (via web UI) move a file and a directory (including some dehydrated files) from one location to another                                                                                                                                                                                                                                                                                                           | The local sync root reflects the move: both hydrated and dehydrated files show up in the new location and removed from the old location                                                                                                                                                                                                                                                                                                         | :construction: Win<br>:construction: macOS<br>:construction: Linux |                               |
```

> | 9  | Remove directory while file open & then edit | 1. Create a directory inside local sync folder; create a file inside and open it in LibreOffice (LO) (file locked) <br>2. On the server, delete/remove the directory <br>3. While LO is still open, edit and save the file <br>4. Close LO                                                                                                                                                                                             | After closing LO, the directory and file are recreated locally (and synced to server)                                                                                                                                                                                                                                                                                                                                                            | :construction: Win<br>:construction: macOS<br>:construction: Linux |                               |

Reviewer: ```suggestion
| 9  | Remove directory on the server while the file is still open locally | 1. Create a directory inside local sync folder; create a file inside and open it in LibreOffice (LO) (file locked) <br>2. On the server, delete/remove the directory <br>3. While LO is still open, edit and save the file <br>4. Close LO                                                                                                                                                                                             | After closing LO, the directory and file are recreated locally (and synced to server)                                                                                                                                                                                                                                                                                                                                                            | :construction: Win<br>:construction: macOS<br>:construction: Linux |                               |
```

| 4  | file versions       | 1. Open the context menu of a synced markdown file <br>2. Click on "Show file versions in web browser"                                                   | A browser window opens and takes the user to the file, with the side bar opened, showind the versions | :construction: Win<br>:construction: macOS<br>:construction: Linux |                               |

### 14. VFS
> [!NOTE]
> **“Available on this device”** = the file is locally present now, but may be removed automatically to free up space. <br>
> **“Always keep on this device”** = you’ve pinned the file locally, so it stays on your device at all times and won’t be evicted automatically. <br>
> **“Hydrated (full) file”** = the file’s full data is present locally (downloaded or created), and you can convert it into a placeholder (dehydrated) by using the ‘Free up space’ action; additionally, the system could later turn it into a placeholder automatically if space is needed. <br>
> **“Dehydrated (placeholder) file”** = the file appears locally but only as a lightweight placeholder with minimal footprint; actual data isn’t downloaded until accessed. <br>

> > The following image demonstrates how the full pinned, full and placeholder file states are shown in File Explorer: <br>

Reviewer: ```suggestion
> **“Dehydrated (placeholder) file”** = the file appears locally but only as a lightweight placeholder with minimal footprint; actual data isn’t downloaded until accessed. <br>
>
> The following image shows the icons for `Always keep on this device`, `Available on this device (Hydrated)` and `Placeholder (Dehydrated)` status in order: <br>
```

| 1  | Add account and connect to server                               | 1. Create two files with some content inside a folder in the server <br>2. Add an account to the desktop client <br>3. Open local sync folder and check status of all resources (2 files and their parent folder)                                                                                                                                                                                                                      | All resources (2 files and their parent folder) are virtual (Status: dehydrated)                                                                                                                                                                                                                                                                                                                                                                 | :construction: Win |                               |

> | 2  | Open a file                                                     | 1. Create a file with some content in the server <br>2. Add an account to the desktop client <br>3. Open local sync folder and open that file <br>4. Create a new file locally in the local sync folder, edit, save and exit it                                                                                                                                                                                                        | Both files are physically available, status: full file (Status: hydrated)                                                                                                                                                                                                                                                                                                                                                                        | :construction: Win |                               |

Reviewer: ```suggestion
| 2  | Open a file                                                     | 1. Create a file with some content in the server <br>2. Add an account to the desktop client <br>3. Open local sync folder and open that file <br>4. Create a new file locally in the local sync folder, edit, save and exit it                                                                                                                                                                                                        | Both files are physically available (Status: hydrated)                                                                                                                                                                                                                                                                                                                                                                        | :construction: Win |                               |
```
